src/main/scala/com.snowplowanalytics.sauna/apis/Sendgrid.scala

```diff
@@ -177,4 +192,382 @@ object Sendgrid {
       case Left(errors) => Left("Cannot extract custom fields information: " + errors.toString)
     }
   }
+
+  /**
+   * Represents an email recipient - may contain the recipient's name,
+   * must contain their email.
+   */
+  case class SendgridEmailObject(
+    email: String,
+    name: Option[String]
+  )
+
+  implicit val sendgridEmailObjectReads: Reads[SendgridEmailObject] = (
+    (JsPath \ "email").read[String] and
+      (JsPath \ "name").readNullable[String]
+    ) (SendgridEmailObject.apply _)
+  implicit val sendgridEmailObjectWrites: Writes[SendgridEmailObject] = Json.writes[SendgridEmailObject]
+
+  /**
+   * Represents a message's "envelope" - metadata containing information
+   * about who should receive an individual message and how it should be handled.
+   */
+  case class SendgridPersonalization(
+    to: Vector[SendgridEmailObject],
```

Comment:
I agree that concrete data structures are preferable, however would choose `List` (I assume we don't need random element access). We already use `Seq` too much in Sauna https://orrsella.com/2014/02/27/the-default-scala-seq-is-mutable-wait-what/